rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -27,12 +26,35 @@
 ADD_LIQUIDITY = b'B?d\x95\xa0\x8f\xc6RB\\\xf4\xed\r\x1f\x9e7\xe5q\xd9\xb9R\x9b\x1c\x1c#\xcc\xe7\x80\xb2\xe7\xdf\r'  # noqa: E501
 ADD_LIQUIDITY_2_ASSETS = b'&\xf5Z\x85\x08\x1d$\x97N\x85\xc6\xc0\x00E\xd0\xf0E9\x91\xe9Xs\xf5+\xff\r!\xaf@y\xa7h'  # noqa: E501
 ADD_LIQUIDITY_4_ASSETS = b'?\x19\x15w^\x0c\x9a8\xa5z{\xb7\xf1\xf9\x00_Ho\xb9\x04\xe1\xf8J\xa2\x156MVs\x19\xa5\x8d'  # noqa: E501
-REMOVE_LIQUIDITY = b"Z\xd0V\xf2\xe2\x8a\x8c\xec# \x15@k\x846h\xc1\xe3l\xdaY\x81'\xec;\x8cY\xb8\xc7's\xa0"  # noqa: E501
-REMOVE_ONE = b'\x9e\x96\xdd;\x99z*%~\xecM\xf9\xbbn\xafbn m\xf5\xf5C\xbd\x966\x82\xd1C0\x0b\xe3\x10'  # noqa: E501
-REMOVE_LIQUIDITY_3_ASSETS = b'\xa4\x9dL\xf0&V\xae\xbf\x8cw\x1fZ\x85\x85c\x8a*\x15\xeel\x97\xcfr\x05\xd4 \x8e\xd7\xc1\xdf%-'  # noqa: E501
-REMOVE_LIQUIDITY_4_ASSETS = b'\x98x\xca7^\x10o*C\xc3\xb5\x99\xfcbEh\x13\x1cL\x9aK\xa6j\x14V7\x15v;\xe9\xd5\x9d'  # noqa: E501
 REMOVE_LIQUIDITY_IMBALANCE = b'\xb9d\xb7/s\xf5\xef[\xf0\xfd\xc5Y\xb2\xfa\xb9\xa7\xb1*9\xe4x\x17\xa5G\xf1\xf0\xae\xe4\x7f\xeb\xd6\x02'  # noqa: E501
-CURVE_Y_DEPOSIT = string_to_evm_address('0xbBC81d23Ea2c3ec7e56D39296F0cbB648873a5d3')
+REMOVE_LIQUIDITY_EVENTS = {
+    REMOVE_LIQUIDITY_IMBALANCE,
+    b'\x9e\x96\xdd;\x99z*%~\xecM\xf9\xbbn\xafbn m\xf5\xf5C\xbd\x966\x82\xd1C0\x0b\xe3\x10',  # REMOVE_ONE  # noqa: E501
+    b"Z\xd0V\xf2\xe2\x8a\x8c\xec# \x15@k\x846h\xc1\xe3l\xdaY\x81'\xec;\x8cY\xb8\xc7's\xa0",  # REMOVE_LIQUIDITY  # noqa: E501,
+    b'\xa4\x9dL\xf0&V\xae\xbf\x8cw\x1fZ\x85\x85c\x8a*\x15\xeel\x97\xcfr\x05\xd4 \x8e\xd7\xc1\xdf%-',  # REMOVE_LIQUIDITY_3_ASSETS  # noqa: E501
+    b'\x98x\xca7^\x10o*C\xc3\xb5\x99\xfcbEh\x13\x1cL\x9aK\xa6j\x14V7\x15v;\xe9\xd5\x9d',  # REMOVE_LIQUIDITY_4_ASSETS  # noqa: E501
+}
+CURVE_DEPOSIT_CONTRACTS = {
```

Comment:
> How did you get this list of deposit contracts?

https://etherscan.io/find-similar-contracts?a=0x094d12e5b541784701FD8d65F11fc0598FBC6332

> Did you make sure that these are the only deposit contracts?

Yeah, used etherscan fuzzy search.
<img width="1316" alt="image" src="https://user-images.githubusercontent.com/72208758/224970930-3650304e-6708-430b-9785-32175a025098.png">
<img width="1345" alt="image" src="https://user-images.githubusercontent.com/72208758/224971610-bf2a5ddf-ecab-4a47-99ba-925e841edf05.png">


> How frequently can new such contracts pop up? Will it worth putting them in an automatically updatable cache?

The last time such contract popped up was Feb, 2021 so I don' think an automatically updateable cache is needed.

Reply:
Makes sense

Reply:
Interesting. Nice approach @prettyirrelevant.

Can you please also do the following:

1. Write how you got the list (as you did in the PR comment), in a comment above the list in the code?
2.Ask in zapper's or whoever makes them chat/discord/twitter if they have maybe a list we can query instead of doing it manually?

Reply:
> Ask in zapper's or whoever makes them chat/discord/twitter if they have maybe a list we can query instead of doing it manually?

The list they provided was for just a subset(like 4). Also, there's no API nor registry contract for this